conclave-common/src/main/kotlin/com/r3/conclave/common/internal/CallInitiator.kt

```diff
@@ -0,0 +1,17 @@
+package com.r3.conclave.common.internal
+
+import java.nio.ByteBuffer
+
+interface CallInitiator<CALL_ID_TYPE> {
```

Comment:
This does not really seem a proper "old style" interface, not sure if we follow any convention though.
But here comes another bigger question: `CallInitiator`, together with `CallAcceptor` are always used together and we only have two combinations for them. We will never have more than 2 combinations I think.
I am questioning if we need this "templatization" ("genericsation", not sure how to call this in Java). 
Maybe we can at least merge them together in one class?

Reply:
I have combined these, it actually works quite well I think!